packages/chips/ChipSet.js

```diff
@@ -23,24 +23,84 @@
 import React, {Component} from 'react';
 import classnames from 'classnames';
 import PropTypes from 'prop-types';
+import {MDCChipSetFoundation} from '@material/chips';
 
 import ChipCheckmark from './ChipCheckmark';
-import Chip from './Chip';
 
 export default class ChipSet extends Component {
   checkmarkWidth_ = 0;
+  foundation_ = null;
+  state = {
+    selectedChipIds: new Set(this.props.selectedChipIds),
+  };
+
+  componentDidMount() {
+    this.foundation_ = new MDCChipSetFoundation(this.adapter);
+    this.foundation_.init();
+    this.updateChipSelection();
+  }
+
+  componentDidUpdate() {
+    this.updateChipSelection();
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
 
   get classes() {
-    const {className} = this.props;
-    return classnames('mdc-chip-set', className);
+    const {className, filter, choice, input} = this.props;
+    return classnames('mdc-chip-set', className, {
+      'mdc-chip-set--filter': filter,
+      'mdc-chip-set--choice': choice,
+      'mdc-chip-set--input': input,
+    });
   }
 
   get adapter() {
     return {
       hasClass: (className) => this.classes.split(' ').includes(className),
+      setSelected: (chipId, selected) => {
+        const {selectedChipIds} = this.state;
+        if (selected) {
+          selectedChipIds.add(chipId);
+        } else {
+          selectedChipIds.delete(chipId);
+        }
+        this.setState({selectedChipIds});
+      },
     };
   }
 
+  updateChipSelection() {
+    React.Children.forEach(this.props.children, (child) => {
+      const {id} = child.props;
+      if (this.props.selectedChipIds.indexOf(id) > -1) {
+        this.foundation_.select(id);
+      } else {
```

Comment:
what if we had a `child.props.selected` check here? So we don't always call deselected if the chip is already deselected.